BeLazy167/stacksplit
# Clerk

# Add Clerk Authentication and UI Improvements

## Changes Overview
- Integrated Clerk authentication system with email verification flow
- Added protected routes with auth state management
- Implemented new tab navigation structure with authenticated routes
- Created new UI screens with Tamagui components:
  - Sign In screen
  - Sign Up screen with email verification
  - Profile management
  - Groups management
  - Expenses tracking

## Key Features
- Complete authentication flow with email verification
- Protected routes with automatic redirects
- Responsive and themed UI components
- Dark mode support
- Profile management with user settings
- Sample expense tracking interface
- Group management system

## Technical Details
- Set up Clerk Provider with secure token storage using expo-secure-store
- Implemented auth state management using Clerk hooks
- Created auth-protected tab layout
- Added sample data structures for expenses and groups
- Integrated Tamagui theming system

## Testing Instructions
1. Set up environment variables:
   ```
   EXPO_PUBLIC_CLERK_PUBLISHABLE_KEY=your_clerk_key
   ```
2. Run the application
3. Test authentication flow:
   - Sign up with email
   - Verify email
   - Sign in
   - Sign out
4. Verify protected routes are working correctly
5. Test UI responsiveness and theme consistency

## Dependencies Added
- @clerk/clerk-expo
- tamagui and related packages
- expo-secure-store
- Additional Expo packages for UI components

## Notes
- Removed Supabase integration in favor of Clerk
- Sample data is currently hardcoded for UI demonstration
- Dark mode implementation is prepared but needs theme switching logic
